src/services/ApiService.ts

```diff
@@ -184,59 +185,79 @@ export class ApiService {
 
         if (preferences) {
             this.upgradePreferences(preferences);
+            console.log(preferences);
             const valid = ApiService.PreferenceValidator(preferences);
+            let deletedKeys: string[] = [];
             if (!valid) {
                 for (const error of ApiService.PreferenceValidator.errors ?? []) {
                     if (error.instancePath) {
                         console.log(`Removing invalid preference ${error.instancePath}`);
                         // Trim the leading "." from the path
-                        delete preferences[error.instancePath.substring(1)];
+                        const errorKey = error.instancePath.substring(1);
+                        // Remove the invalid preference from the preferences object
+                        delete preferences[errorKey];
+                        this.clearPreferences([errorKey]);
+                        deletedKeys.push(errorKey);
                     }
                 }
+
+                if (deletedKeys.length > 0) {
+                    // Show an alert to the user about the deleted preferences
+                    AppStore.Instance.alertStore.showAlert(`Deleted invalid preferences: ${deletedKeys.join(", ")}`);
```

Comment:
Minor point: instead of "deleted" here I would say: `Invalid preferences reset to defaults: ...`